
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>pip_services3_commons.errors package &#8212; Pip.Services Commons for Python API</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="pip_services3_commons.errors.ApplicationException module" href="pip_services3_commons.errors.ApplicationException.html" />
    <link rel="prev" title="pip_services3_commons.data.StringValueMap module" href="pip_services3_commons.data.StringValueMap.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="pip_services3_commons.errors.ApplicationException.html" title="pip_services3_commons.errors.ApplicationException module"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="pip_services3_commons.data.StringValueMap.html" title="pip_services3_commons.data.StringValueMap module"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Pip.Services Commons for Python API</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="modules.html" >pip_services3_commons</a> &#187;</li>
          <li class="nav-item nav-item-2"><a href="pip_services3_commons.html" accesskey="U">pip_services3_commons package</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="pip-services3-commons-errors-package">
<h1>pip_services3_commons.errors package<a class="headerlink" href="#pip-services3-commons-errors-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html">pip_services3_commons.errors.ApplicationException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip-services3-commons-errors-applicationexception">pip_services3_commons.errors.ApplicationException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ApplicationExceptionFactory.html">pip_services3_commons.errors.ApplicationExceptionFactory module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ApplicationExceptionFactory.html#pip-services3-commons-errors-applicationexceptionfactory">pip_services3_commons.errors.ApplicationExceptionFactory</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.BadRequestException.html">pip_services3_commons.errors.BadRequestException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.BadRequestException.html#pip-services-common-errors-badrequestexception">pip_services_common.errors.BadRequestException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ConfigException.html">pip_services3_commons.errors.ConfigException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ConfigException.html#pip-services-common-errors-configexception">pip_services_common.errors.ConfigException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ConflictException.html">pip_services3_commons.errors.ConflictException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ConflictException.html#pip-services-common-errors-conflictexception">pip_services_common.errors.ConflictException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ConnectionException.html">pip_services3_commons.errors.ConnectionException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ConnectionException.html#pip-services-common-errors-connectionexception">pip_services_common.errors.ConnectionException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ErrorCategory.html">pip_services3_commons.errors.ErrorCategory module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ErrorCategory.html#pip-services3-commons-errors-errorcategory">pip_services3_commons.errors.ErrorCategory</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ErrorDescription.html">pip_services3_commons.errors.ErrorDescription module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ErrorDescription.html#pip-services3-commons-errors-errordescription">pip_services3_commons.errors.ErrorDescription</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.ErrorDescriptionFactory.html">pip_services3_commons.errors.ErrorDescriptionFactory module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.ErrorDescriptionFactory.html#pip-services3-commons-errors-errordescriptionfactory">pip_services3_commons.errors.ErrorDescriptionFactory</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.FileException.html">pip_services3_commons.errors.FileException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.FileException.html#pip-services-common-errors-fileexception">pip_services_common.errors.FileException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.InternalException.html">pip_services3_commons.errors.InternalException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.InternalException.html#pip-services-common-errors-internalexception">pip_services_common.errors.InternalException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.InvalidStateException.html">pip_services3_commons.errors.InvalidStateException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.InvalidStateException.html#pip-services-common-errors-invalidstateexception">pip_services_common.errors.InvalidStateException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.InvocationException.html">pip_services3_commons.errors.InvocationException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.InvocationException.html#pip-services-common-errors-invocationexception">pip_services_common.errors.InvocationException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.NotFoundException.html">pip_services3_commons.errors.NotFoundException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.NotFoundException.html#pip-services-common-errors-notfoundexception">pip_services_common.errors.NotFoundException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.UnauthorizedException.html">pip_services3_commons.errors.UnauthorizedException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.UnauthorizedException.html#pip-services-common-errors-unauthorizedexception">pip_services_common.errors.UnauthorizedException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.UnknownException.html">pip_services3_commons.errors.UnknownException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.UnknownException.html#pip-services-common-errors-unknownexception">pip_services_common.errors.UnknownException</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pip_services3_commons.errors.UnsupportedException.html">pip_services3_commons.errors.UnsupportedException module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pip_services3_commons.errors.UnsupportedException.html#pip-services-common-errors-unsupportedexception">pip_services_common.errors.UnsupportedException</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="module-pip_services3_commons.errors">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-pip_services3_commons.errors" title="Permalink to this headline">¶</a></h2>
<div class="section" id="pip-services3-commons-errors-init">
<h3>pip_services3_commons.errors.__init__<a class="headerlink" href="#pip-services3-commons-errors-init" title="Permalink to this headline">¶</a></h3>
<p>Portable and localizable Exceptions classes. Each Exception, in addition to a description
and stack trace has a unique string code, details array (which can be used for creating localized strings).</p>
<p>Way to use:
  <ul>
<li>An existing exception class can be used.
    <li>A child class that extends ApplicationException can we written.
        <li>A exception can be wrapped around (into?) an existing application exception.</p>
  </ul>
<p>Exceptions are serializable. The exception classes themselves are not serializable, but
they can be converted to ErrorDescriptions, which are serializable in one language, transferred
to the receiving side, and deserialized in another language. After deserialization, the initial
exception class can be restored.</p>
<p>Additionally: when transferring an exception from one language to another, the exception type
that is closest to the initial exception type is chosen from the exceptions available in the target language.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">copyright:</th><td class="field-body">Conceptual Vision Consulting LLC 2018-2019, see AUTHORS for more details.</td>
</tr>
<tr class="field-even field"><th class="field-name">license:</th><td class="field-body">MIT, see LICENSE for more details.</td>
</tr>
</tbody>
</table>
<dl class="class">
<dt id="pip_services3_commons.errors.ErrorCategory">
<em class="property">class </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ErrorCategory</code><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Defines standard error categories to application exceptions supported by PipServices toolkit.</p>
<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.BadRequest">
<code class="descname">BadRequest</code><em class="property"> = 'BadRequest'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.BadRequest" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors due to improper user requests, like
missing or wrong parameters</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Conflict">
<code class="descname">Conflict</code><em class="property"> = 'Conflict'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Conflict" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors raised by conflict in object versions
posted by user and stored on server.</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.FailedInvocation">
<code class="descname">FailedInvocation</code><em class="property"> = 'FailedInvocation'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.FailedInvocation" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors returned by remote services or network
during call attempts</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.FileError">
<code class="descname">FileError</code><em class="property"> = 'FileError'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.FileError" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors in read/write file operations</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Internal">
<code class="descname">Internal</code><em class="property"> = 'Internal'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Internal" title="Permalink to this definition">¶</a></dt>
<dd><p>Internal errors caused by programming mistakes</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.InvalidState">
<code class="descname">InvalidState</code><em class="property"> = 'InvalidState'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.InvalidState" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors related to operations called in wrong component state.
For instance, business calls when component is not ready</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Misconfiguration">
<code class="descname">Misconfiguration</code><em class="property"> = 'Misconfiguration'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Misconfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors related to mistakes in user-defined configuration</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.NoResponse">
<code class="descname">NoResponse</code><em class="property"> = 'NoResponse'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.NoResponse" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors happened during connection to remote services.
They can be related to misconfiguration, network issues
or remote service itself</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.NotFound">
<code class="descname">NotFound</code><em class="property"> = 'NotFound'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.NotFound" title="Permalink to this definition">¶</a></dt>
<dd><p>Error caused by attempt to access missing object</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Unauthorized">
<code class="descname">Unauthorized</code><em class="property"> = 'Unauthorized'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Unauthorized" title="Permalink to this definition">¶</a></dt>
<dd><p>Access errors caused by missing user identity
or security permissions</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Unknown">
<code class="descname">Unknown</code><em class="property"> = 'Unknown'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Unknown" title="Permalink to this definition">¶</a></dt>
<dd><p>Unknown or unexpected errors</p>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorCategory.Unsupported">
<code class="descname">Unsupported</code><em class="property"> = 'Unsupported'</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorCategory.Unsupported" title="Permalink to this definition">¶</a></dt>
<dd><p>Errors caused by calls to unsupported
or not yet implemented functionality</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pip_services3_commons.errors.ErrorDescription">
<em class="property">class </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ErrorDescription</code><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Serializeable error description. It is use to pass information about errors
between microservices implemented in different languages. On the receiving side
ErrorDescription is used to recreate exception object close to its original type
without missing additional details.</p>
<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.category">
<code class="descname">category</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.category" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.cause">
<code class="descname">cause</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.cause" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.code">
<code class="descname">code</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.code" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.correlation_id">
<code class="descname">correlation_id</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.correlation_id" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.details">
<code class="descname">details</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.details" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pip_services3_commons.errors.ErrorDescription.from_json">
<em class="property">static </em><code class="descname">from_json</code><span class="sig-paren">(</span><em>json</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.from_json" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.message">
<code class="descname">message</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.message" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.stack_trace">
<code class="descname">stack_trace</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.stack_trace" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ErrorDescription.status">
<code class="descname">status</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.status" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ErrorDescription.to_json">
<code class="descname">to_json</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescription.to_json" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.ApplicationException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ApplicationException</code><span class="sig-paren">(</span><em>category='Unknown'</em>, <em>correlation_id=None</em>, <em>code='UNKNOWN'</em>, <em>message='Unknown error'</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>Defines a base class to defive various application exceptions.</p>
<p>Most languages have own definition of base exception (error) types.
However, this class is implemented symmetrically in all languages
supported by PipServices toolkit. It allows to create portable implementations
and support proper error propagation in microservices calls.</p>
<p>Error propagation means that when microservice implemented in one language
calls microservice(s) implemented in a different language(s), errors are returned
throught the entire call chain and restored in their original (or close) type.</p>
<p>Since number of potential exception types is endless, PipServices toolkit
supports only 12 standard categories of exceptions defined in ErrorCategory.
This ApplicationException class acts as a basis for all other 12 standard exception types.</p>
<p>Most exceptions have just free-form message that describes occured error.
That may not be sufficient to create meaninful error descriptions.
The ApplicationException class proposes an extended error definition that has more standard fields:
<ul>
<li>message: is a human-readable error description
  <li>category: one of 12 standard error categories of errors
  <li>status: numeric HTTP status code for REST invocations
      <li>code: a unique error code, usually defined as “MY_ERROR_CODE”
          <li>correlation_id: a unique transaction id to trace execution through a call chain
              <li>details: map with error parameters that can help to recreate meaningful error description in other languages
                  <li>stack_trace: a stack trace
                      <li>cause: original error that is wrapped by this exception</p>
                      </ul>
<p>ApplicationException class is not serializable. To pass errors through the wire
it is converted into ErrorDescription object and restored on receiving end into identical exception type.</p>
<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.category">
<code class="descname">category</code><em class="property"> = 'Unknown'</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.category" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.cause">
<code class="descname">cause</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.cause" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.code">
<code class="descname">code</code><em class="property"> = 'UNKNOWN'</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.code" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.correlation_id">
<code class="descname">correlation_id</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.correlation_id" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.details">
<code class="descname">details</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.details" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.get_cause_string">
<code class="descname">get_cause_string</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.get_cause_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets original error wrapped by this exception as a string message.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an original error message.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.get_stack_trace_string">
<code class="descname">get_stack_trace_string</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.get_stack_trace_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets a stack trace where this exception occured.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a stack trace as a string.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.message">
<code class="descname">message</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.message" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.set_cause_string">
<code class="descname">set_cause_string</code><span class="sig-paren">(</span><em>value</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.set_cause_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets original error wrapped by this exception as a string message.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>value</strong> – an original error message.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.set_stack_trace_string">
<code class="descname">set_stack_trace_string</code><span class="sig-paren">(</span><em>value</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.set_stack_trace_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a stack trace where this exception occured.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>value</strong> – a stack trace as a string</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.stack_trace">
<code class="descname">stack_trace</code><em class="property"> = None</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.stack_trace" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pip_services3_commons.errors.ApplicationException.status">
<code class="descname">status</code><em class="property"> = 500</em><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.status" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.to_json">
<code class="descname">to_json</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.to_json" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.with_cause">
<code class="descname">with_cause</code><span class="sig-paren">(</span><em>cause</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.with_cause" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a original error wrapped by this exception</p>
<p>This method returns reference to this exception to implement Builder pattern to chain additional calls.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cause</strong> – original error object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">this exception object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.with_code">
<code class="descname">with_code</code><span class="sig-paren">(</span><em>code</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.with_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a unique error code.
This method returns reference to this exception to implement Builder pattern to chain additional calls.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>code</strong> – a unique error code</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">this exception object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.with_correlation_id">
<code class="descname">with_correlation_id</code><span class="sig-paren">(</span><em>correlation_id</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.with_correlation_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a correlation id which can be used to trace this error through a call chain.</p>
<p>This method returns reference to this exception to implement Builder pattern to chain additional calls.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>correlation_id</strong> – a unique transaction id to trace error through call chain</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">this exception object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.with_details">
<code class="descname">with_details</code><span class="sig-paren">(</span><em>key</em>, <em>value</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.with_details" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a parameter for additional error details.
This details can be used to restore error description in other languages.</p>
<p>This method returns reference to this exception to implement Builder pattern to chain additional calls.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>key</strong> – a details parameter name</li>
<li><strong>value</strong> – a details parameter name</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">this exception object</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.with_status">
<code class="descname">with_status</code><span class="sig-paren">(</span><em>status</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.with_status" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets a HTTP status code which shall be returned by REST calls.
This method returns reference to this exception to implement Builder pattern to chain additional calls.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>status</strong> – an HTTP error code.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">this exception object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pip_services3_commons.errors.ApplicationException.wrap">
<code class="descname">wrap</code><span class="sig-paren">(</span><em>cause</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.wrap" title="Permalink to this definition">¶</a></dt>
<dd><p>Wraps another exception into an application exception object.</p>
<p>If original exception is of ApplicationException type it is returned without changes.
Otherwise a new ApplicationException is created and original error is set as its cause.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cause</strong> – an original error object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">an original or newly created ApplicationException</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pip_services3_commons.errors.ApplicationException.wrap_exception">
<em class="property">static </em><code class="descname">wrap_exception</code><span class="sig-paren">(</span><em>exception</em>, <em>cause</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationException.wrap_exception" title="Permalink to this definition">¶</a></dt>
<dd><p>Wraps another exception into specified application exception object.</p>
<p>If original exception is of ApplicationException type it is returned without changes.
Otherwise the original error is set as a cause to specified ApplicationException object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>exception</strong> – an ApplicationException object to wrap the cause</li>
<li><strong>cause</strong> – an original error object</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">an original or newly created ApplicationException</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pip_services3_commons.errors.ApplicationExceptionFactory">
<em class="property">class </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ApplicationExceptionFactory</code><a class="headerlink" href="#pip_services3_commons.errors.ApplicationExceptionFactory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Factory to recreate exceptions from ErrorDescription values passed through the wire.</p>
<dl class="staticmethod">
<dt id="pip_services3_commons.errors.ApplicationExceptionFactory.create">
<em class="property">static </em><code class="descname">create</code><span class="sig-paren">(</span><em>description</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ApplicationExceptionFactory.create" title="Permalink to this definition">¶</a></dt>
<dd><p>Recreates ApplicationException object from serialized ErrorDescription.</p>
<p>It tries to restore original exception type using type or error category fields.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>description</strong> – a serialized error description received as a result of remote call</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">ApplicationException object from serialized ErrorDescription.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pip_services3_commons.errors.ErrorDescriptionFactory">
<em class="property">class </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ErrorDescriptionFactory</code><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescriptionFactory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Factory to create serializeable ErrorDescription from ApplicationException
or from arbitrary errors.</p>
<p>The ErrorDescriptions are used to pass errors through the wire between microservices
implemented in different languages. They allow to restore exceptions on the receiving side
close to the original type and preserve additional information.</p>
<dl class="staticmethod">
<dt id="pip_services3_commons.errors.ErrorDescriptionFactory.create">
<em class="property">static </em><code class="descname">create</code><span class="sig-paren">(</span><em>ex</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ErrorDescriptionFactory.create" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a serializable ErrorDescription from error object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ex</strong> – an error object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a serializeable ErrorDescription object that describes the error.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.UnknownException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">UnknownException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.UnknownException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Unknown or unexpected errors</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.InternalException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">InternalException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.InternalException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors caused by programming mistakes</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.ConfigException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ConfigException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ConfigException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors related to mistakes in the microservice’s user-defined configurations.</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.InvalidStateException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">InvalidStateException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.InvalidStateException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors related to calling operations, which require the component to be in a specific state.</p>
<p>For instance, business calls when component is not ready</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.ConnectionException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ConnectionException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ConnectionException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors happened during connection to remote services.
They can be related to misconfiguration, network issues
or remote service itself</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.InvocationException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">InvocationException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.InvocationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors returned by remote services or by the network during call attempts.</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.FileException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">FileException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.FileException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors in read/write local disk operations.</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.BadRequestException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">BadRequestException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.BadRequestException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors due to improper user requests.</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.NotFoundException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">NotFoundException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.NotFoundException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Error caused by attempt to access missing object</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.UnauthorizedException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">UnauthorizedException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.UnauthorizedException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Access errors caused by missing user identity (authentication error)
or incorrect security permissions (authorization error).</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.ConflictException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">ConflictException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.ConflictException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors raised by conflicts between object versions that were
posted by the user and those that are stored on the server.</p>
</dd></dl>

<dl class="exception">
<dt id="pip_services3_commons.errors.UnsupportedException">
<em class="property">exception </em><code class="descclassname">pip_services3_commons.errors.</code><code class="descname">UnsupportedException</code><span class="sig-paren">(</span><em>correlation_id=None</em>, <em>code=None</em>, <em>message=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pip_services3_commons.errors.UnsupportedException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="pip_services3_commons.errors.ApplicationException.html#pip_services3_commons.errors.ApplicationException.ApplicationException" title="pip_services3_commons.errors.ApplicationException.ApplicationException"><code class="xref py py-class docutils literal notranslate"><span class="pre">pip_services3_commons.errors.ApplicationException.ApplicationException</span></code></a></p>
<p>Errors caused by calls to unsupported or not yet implemented functionality</p>
</dd></dl>

</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">pip_services3_commons.errors package</a><ul>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-pip_services3_commons.errors">Module contents</a><ul>
<li><a class="reference internal" href="#pip-services3-commons-errors-init">pip_services3_commons.errors.__init__</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="pip_services3_commons.data.StringValueMap.html"
                        title="previous chapter">pip_services3_commons.data.StringValueMap module</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="pip_services3_commons.errors.ApplicationException.html"
                        title="next chapter">pip_services3_commons.errors.ApplicationException module</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/pip_services3_commons.errors.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="pip_services3_commons.errors.ApplicationException.html" title="pip_services3_commons.errors.ApplicationException module"
             >next</a> |</li>
        <li class="right" >
          <a href="pip_services3_commons.data.StringValueMap.html" title="pip_services3_commons.data.StringValueMap module"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Pip.Services Commons for Python API</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="modules.html" >pip_services3_commons</a> &#187;</li>
          <li class="nav-item nav-item-2"><a href="pip_services3_commons.html" >pip_services3_commons package</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Conceptual Vision Consulting LLC.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.1.
    </div>
  </body>
</html>